src/main/java/com/google/api/codegen/gapic/GapicProviderFactory.java

```diff
@@ -15,12 +15,16 @@
 package com.google.api.codegen.gapic;
 
 import com.google.api.codegen.config.ApiConfig;
+import com.google.api.codegen.config.PackageMetadataConfig;
 import com.google.api.tools.framework.model.Model;
 import java.util.List;
 
 /** A factory for GapicProviders which perform code generation. */
 public interface GapicProviderFactory<ProviderT> {
   /** Create the provider from the given model, apiConfig, and generatorConfig. */
   public List<ProviderT> create(
-      Model model, ApiConfig apiConfig, GapicGeneratorConfig generatorConfig);
+      Model model,
+      ApiConfig apiConfig,
+      GapicGeneratorConfig generatorConfig,
+      PackageMetadataConfig packageConfig);
```

Comment:
Does it make sense to put PackageMetadataConfig under GapicGeneratorConfig? I am a bit cautious of adding more parameters to this interface and conceptually package metadata is a part of GAPIC gen now.